packages/chain-adapters/src/solana/SolanaChainAdapter.ts

```diff
@@ -380,32 +384,40 @@ export class ChainAdapter implements IChainAdapter<KnownChainIds.SolanaMainnet>
   ): Promise<FeeDataEstimate<KnownChainIds.SolanaMainnet>> {
     try {
       const { baseFee, fast, average, slow } = await this.providers.http.getPriorityFees()
+      const { sendMax, chainSpecific } = input
+      const { instructions } = chainSpecific
 
       const serializedTx = await this.buildEstimationSerializedTx(input)
-      const computeUnits = await this.providers.http.estimateFees({
+      const baseComputeUnits = await this.providers.http.estimateFees({
         estimateFeesBody: { serializedTx },
       })
 
+      const computeUnits = sendMax
+        ? bnOrZero(baseComputeUnits).times(SOLANA_COMPUTE_UNITS_BUFFER_MULTIPLIER).toFixed()
+        : baseComputeUnits
```

Comment:
We add a buffer to the compute units, specially for sendMax operations

Other part of the app should be adding the buffer depending on their logic, they might not need it